-Wdeprecated-copy compile error in call_op_set.h when using clang 10 or above

### What version of gRPC and what language are you using?

`master`

### What operating system (Linux, Windows,...) and version?

Linux Ubuntu 18.04 kernel 4.19

### What runtime / compiler are you using (e.g. python version or version of gcc)

clang 10

### What did you do?

Compile anything that uses `grpcpp/impl/codegen/call_op_set.h`, e.g. by including `async_generic_service.h`.

### What did you expect to see?

Compiles fine.

### What did you see instead?

```
grpc/include/grpcpp/impl/codegen/call_op_set.h:88:3: error: definition of implicit copy assignment operator for 'WriteOptions' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  WriteOptions(const WriteOptions& other)
  ^
```

--- a/include/grpcpp/impl/codegen/call_op_set.h
+++ b/include/grpcpp/impl/codegen/call_op_set.h
@@ -79,11 +79,6 @@ inline grpc_metadata* FillMetadataArray(
 class WriteOptions {
  public:
   WriteOptions() : flags_(0), last_message_(false) {}
-  WriteOptions(const WriteOptions& other)
-      : flags_(other.flags_), last_message_(other.last_message_) {}
-
-  /// Default assignment operator
-  WriteOptions& operator=(const WriteOptions& other) = default;
 
   /// Clear all flags.
   inline void Clear() { flags_ = 0; }

